pkg/operation/botanist/component/kubeapiserverexposure/kube_apiserver_sni.go

```diff
@@ -214,6 +218,19 @@ func (s *sni) Deploy(ctx context.Context) error {
 		accessControl.Spec = istioapisecurityv1beta1.AuthorizationPolicy{
 			Action: accessControlSpec.Action,
 			Rules:  accessControlSpec.Rules,
+			Selector: &istiov1beta1.WorkloadSelector{
+				MatchLabels: map[string]string{
+					"app":   "istio-ingressgateway",
+					"istio": "ingressgateway",
+				},
+			},
+		}
+
+		for i := range accessControl.Spec.Rules {
+			accessControl.Spec.Rules[i].When = []*istioapisecurityv1beta1.Condition{{
+				Key:    "connection.sni",
+				Values: s.values.Hosts,
```

Comment:
This is actually very nice. I am wondering if it's a good idea to exclude the internal API server dns name from here to prevent Gardener from reaching it's API server. Not sure if it's a safe solution in general to exclude this.

Reply:
We also thought about this. If you would exclude the internal Host you would need to create another ALLOW rule which is always active that only contains the internal URL.
The problem we faced is that in our case customers could also access via the internal domain (at least thats what i think for now) so we thought its better to apply the rules for everything.
But i think that topic needs to be discussed since we also need to figured out what needs to be allowed for Gardener internal traffic :)